dipy/data/fetcher.py

```diff
@@ -1786,3 +1783,105 @@ def fetch_hcp(subjects,
                            "PipelineDescription": {'Name': 'hcp_pipeline'}})
 
     return data_files, pjoin(my_path, study)
+
+
+def fetch_hbn(subjects, path=None):
+    """
+    Fetches preprocessed data from the Healthy Brain Network POD2 study [1, 2]_.
+
+    Parameters
+    ----------
+    subjects : list
+        Identifiers of the subjects to download.
+        For example: ["NDARAA112DMH", "NDARAA117NEJ"].
+    path : string, optional
+        Path to save files into. Default: '~/AFQ_data'
+
+    Returns
+    -------
+    dict with remote and local names of these files,
+    path to BIDS derivative dataset
+
+    Notes
+    -----
+
+    .. [1] Alexander LM, Escalera J, Ai L, et al. An open resource for
+        transdiagnostic research in pediatric mental health and learning
+        disorders. Sci Data. 2017;4:170181.
+
+    .. [2] Richie-Halford A, Cieslak M, Ai L, et al. An analysis-ready and
+        quality controlled resource for pediatric brain white-matter research.
+        Scientific Data. 2022;9(1):1-27.
+
+    """
+
+    if has_boto3:
+        from botocore import UNSIGNED
+        from botocore.client import Config
+    else:
+        TripWire("The `fetch_hbn` function requires the boto3",
+                 "library, but that is not installed.")
+
+    # Anonymous access:
+    client = boto3.client('s3', config=Config(signature_version=UNSIGNED))
+
+    if path is None:
+        if not op.exists(dipy_home):
+            os.mkdir(dipy_home)
+        my_path = dipy_home
+    else:
+        my_path = path
+
+    base_dir = op.join(my_path, "HBN", 'derivatives', 'qsiprep')
+
+    if not os.path.exists(base_dir):
+        os.makedirs(base_dir, exist_ok=True)
+
+    data_files = {}
+
+    # If user provided incorrect input, these are typical failures that
+    # are easy to recover from:
+    if isinstance(subjects, int) or isinstance(subjects, str):
+        subjects = [subjects]
+
+    for subject in subjects:
+        initial_query = client.list_objects(
+            Bucket="fcp-indi",
+            Prefix=f"data/Projects/HBN/BIDS_curated/sub-{subject}/")
+        ses = initial_query['Contents'][0]["Key"].split('/')[5]
```

Comment:
To handle potential issue, 
I recommend using  
` ses = initial_query.get('Contents', [])` and check if everything is ok.

Reply:
Addressed in 7e6153ac5